tests/unmanaged_models/tests.py

```diff
@@ -59,3 +61,11 @@ def test_many_to_many_between_unmanaged_and_managed(self):
         table = Managed1._meta.get_field('mm').m2m_db_table()
         tables = connection.introspection.table_names()
         self.assertIn(table, tables, "Table '%s' does not exist." % table)
+
+    def test_unmanaged_many_to_many_between_managed_models(self):
+        from django.db import connection
+        intermediate_table = UnmanagedIntermediate._meta.db_table
+        connection.cursor().execute('CREATE OR REPLACE VIEW {} AS '
+                                    'SELECT 1 AS id, 1 AS d01_id, 1 AS d02_id;'.format(intermediate_table))
```

Comment:
Unless I'm mistaken you don't need to create a view to hit the bug; calling `django_table_names` with `only_existing=False` (the default value) should also fail before the patch is applied.